packages/devtools_app/lib/src/timeline/flutter/flutter_frames_chart.dart

```diff
@@ -28,34 +27,82 @@ import 'package:mp_chart/mp/core/utils/painter_utils.dart';
 import 'package:mp_chart/mp/core/value_formatter/default_value_formatter.dart';
 import 'package:mp_chart/mp/core/value_formatter/value_formatter.dart';
 
+import '../../flutter/controllers.dart';
+import '../../ui/fake_flutter/_real_flutter.dart';
+import '../timeline_controller.dart';
+import '../timeline_model.dart';
+
 class FlutterFramesChart extends StatefulWidget {
   const FlutterFramesChart();
 
   @override
-  FlutterFramesChartState createState() => FlutterFramesChartState();
+  _FlutterFramesChartState createState() => _FlutterFramesChartState();
 }
 
-class FlutterFramesChartState extends State<FlutterFramesChart>
+class _FlutterFramesChartState extends State<FlutterFramesChart>
     implements OnChartValueSelectedListener {
-  BarChartController _chartController;
+  TimelineController _controller;
 
-  BarChartController get chartController => _chartController;
+  List<TimelineFrame> frames = [];
 
-  /// Index into the raw data.
-  int timerDataIndex;
+  BarChartController _chartController;
 
-  /// Active timer running.
-  Timer _timer;
+  BarChartController get chartController => _chartController;
 
   /// Datapoint entry for each frame duration (UI/GPU) for stacked bars.
   final List<BarEntry> _frameDurations = <BarEntry>[];
 
   /// Set of all duration information (the data, colors, etc).
   BarDataSet frameDurationsSet;
 
+  final int totalFramesToChart = 150;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _controller = Controllers.of(context).timeline;
+
+    // Process each timeline frame.
+    _controller.frameBasedTimeline.onFrameAdded.listen((extraFrames) {
```

Comment:
rename `extraFrames` to `frame` or `newFrame` since it is singular

Reply:
newFrame done.